sdk/bare/common/sys/transform.c

```diff
@@ -6,7 +6,12 @@ void transform_clarke(double C, double *abc, double *xyz)
 {
     xyz[0] = C * (1 * abc[0] - 0.5 * abc[1] - 0.5 * abc[2]);
     xyz[1] = C * (0 * abc[0] + SQRT3 / 2 * abc[1] - SQRT3 / 2 * abc[2]);
-    xyz[2] = (1 / 3 * abc[0] + 1 / 3 * abc[1] + 1 / 3 * abc[2]);
+    if (C == TRANS_DQZ_C_INVARIANT_AMPLITUDE) {
```

Comment:
@ASHAD1 I believe you are comparing a double to another double here. This is a bad idea due to rounding errors; doubles are basically never exactly equal to each other.

I recommend you create an `enum` typedef where the user specifies the correct enum value and the code takes this and uses the right `C` value internally.

For example:
```c
typedef enum {
  TRANS_DQZ_C_INVARIANT_AMPLITUDE = 0,
  TRANS_DQZ_C_INVARIANT_POWER
} transform_dqz_type_e;
```

The above typedef would go in the header file.